crates/bevy_reflect/src/impls/glam.rs

```diff
@@ -1,44 +1,195 @@
 use crate as bevy_reflect;
+use crate::reflect::Reflect;
 use crate::ReflectDeserialize;
-use bevy_reflect_derive::{impl_from_reflect_value, impl_reflect_value};
+use bevy_reflect_derive::{
+    impl_from_reflect_value, impl_reflect_struct_and_from_reflect_struct, impl_reflect_value,
+};
 use glam::*;
 
-impl_reflect_value!(IVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(IVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(IVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec3A(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Mat3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Mat4(PartialEq, Serialize, Deserialize));
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct IVec2 {
+        x: i32,
+        y: i32
+    }
```

Comment:
This is an interesting way of doing this. It makes sense, but it's not super clear (especially for potential users) that it's not creating the struct, just using it.

I think this would also make a good attribute candidate on its own: [`#[reflect(remote = "MockIVec2")]`](https://serde.rs/remote-derive.html).

Reply:
I'm a bit conflicted on this. I do agree that it's not super clear but adding a remote and a Mock{Type} would expand it significantly and be a bit awkward to use. Maybe something more like this?
```
impl_reflect_struct_and_from_reflect_struct!(
    #[reflect(target = "IVec2", ctor/path, traits)
    struct MockIVec2 {
        x: i32,
        y: i32
    }
)
```
Do you think this is better?

Reply:
Yeah perhaps you're right. I never liked serde's remote approach either. I think your method with the target and stuff is probably more intuitive. Would it make sense to rename this macro to something like `mock_reflect_struct` since it's really only useful for that purpose?

One thing I don't like about this is the use of `target` in the `#[reflect]` since it might confuse users into thinking this is available outside this macro (which I don't think it is or will be). What do you think of a syntax like:

```rust
mock_reflect_struct! {
  #[mock = "IVec2"] // <- Required mock target
  #[reflect(ctor/path, traits)] // <- These should probably be optional
  struct {
    x: i32,
    y: i32
  } // <- Actual mock structure
}
```

Or something like that if you have any ideas. Or do you think this isn't the way to go here?

Reply:
I think a rename is a good idea. The current name is descriptive but irritatingly long. However, this would also diverge further from `impl_reflect_value` and `impl_from_reflect_value`, which this is an alternative to and I would prefer to mirror as much as possible (unfortunately splitting the base impl and the from_reflect impl would require duplicating a lot of unnecessary code, so I merged them). I'm not certain that the separate mock attribute is better than a target attribute, I think it would just be more complicated. I do like removing the struct name, but that would also break the DeriveInput parser. That would be a real pain to put back together. Currently I believe the #[reflect] attribute is entirely optional, if you don't have anything to add to it.
And just to be clear, no, the mock struct is never actually constructed, it's just a description of what to pretend the target struct is defined as

Reply:
> I think a rename is a good idea. The current name is descriptive but irritatingly long. However, this would also diverge further from impl_reflect_value and impl_from_reflect_value, which this is an alternative to and I would prefer to mirror as much as possible (unfortunately splitting the base impl and the from_reflect impl would require duplicating a lot of unnecessary code, so I merged them). 

That makes sense. Yeah my reasoning for a rename (besides mocking being the main purpose it seems) was to also imply that we're not creating the struct, just using it. But maybe that's just not easily conveyed no matter the name (so we just need good documentation on the macro function to inform users of that). I’m still in favor of the mock naming, but if you have other ideas or pushback on that, we can bikeshed it a bit more.

> I do like removing the struct name, but that would also break the DeriveInput parser. That would be a real pain to put back together.

And yeah I realized after writing that that keeping a name on the struct would just be easier to parse so I think it's fine to keep it.

> I'm not certain that the separate mock attribute is better than a target attribute, I think it would just be more complicated.

You’re right that it just adds complexity. I think you've convinced me that the way you had it originally is probably the simplest:

```rust
mock_reflect_struct! { // Or whatever we end up calling it
  #[reflect(stuff)] // <- Still optional
  struct IVec2 { // Actual target name used
    x: i32,
    y: i32,
  }
}
```

Sorry for the loop around haha but I’m glad we discussed alternatives at the very least!